app/api/images/route.ts

```diff
@@ -37,15 +37,20 @@ export async function POST(request: Request) {
 
   // Create the image record in Xata
   const record = await xata.db.image.create({
-    name: name as string,
-    image: {
-      name: fileName,
-      mediaType: mimeType,
-      // Xata expects a base64 encoded string for the file content
-      base64Content: fileData.toString('base64')
-    }
+    name: name as string
   });
 
+  // Upload the file and attach it to the image record
+  await xata.files.upload(
+    {
+      table: 'image',
+      column: 'image',
+      record: record.id
+    },
+    fileData
```

Comment:
We are creating a Buffer from an ArrayBuffer some lines before, we could leave it as an array buffer and do

XataFile.fromArrayBuffer(data, { mediaType })

Reply:
I tried this, but I don't think XataFile is meant to be used with binary upload. It resulted in a malformed item being uploaded. Maybe I got the syntax wrong? Sorry, the TS here in a bunch of these objects isn't very helpful.

Also, if we're supposed to be using `XataFile`, we should likely have a lot of this stuff in our docs. I haven't really been aware I should be using it at all.

![image](https://github.com/xataio/sample-nextjs-chakra-gallery-app/assets/324519/3eb652f2-e968-40f1-bec3-cc3b2b572dbc)

Reply:
You can do `xataFile.toBlob()`